mysql-test/suite/innodb/t/temp_table_encrypt.test

```diff
@@ -0,0 +1,93 @@
+CREATE TEMPORARY TABLE t05 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='y';
+
+CREATE TEMPORARY TABLE t06 (a INT) ENGINE=InnoDB ROW_FORMAT=COMPRESSED ENCRYPTION='n';
+INSERT INTO t06 VALUES (1), (2), (3);
+
+# test that we can turn encryption OFF and ON
+
+SET GLOBAL innodb_temp_tablespace_encrypt = OFF;
+
+CREATE TEMPORARY TABLE t07 (a INT) ENGINE=InnoDB;
+INSERT INTO t07 VALUES (1), (2), (3);
+
+SET GLOBAL innodb_temp_tablespace_encrypt = ON;
+
+# make sure encryption cannot be be turned ON/OFF for session
+--error ER_GLOBAL_VARIABLE
+SET SESSION innodb_temp_tablespace_encrypt = OFF;
+--error ER_GLOBAL_VARIABLE
+SET SESSION innodb_temp_tablespace_encrypt = ON;
+
+CREATE TABLE t10 (a INT AUTO_INCREMENT PRIMARY KEY, b INT);
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+INSERT INTO t10 (b) VALUES (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000)), (FLOOR(RAND() * 10000));
+
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
+INSERT INTO t10 (b) SELECT b FROM t10;
+
+CREATE INDEX t10_b ON t10 (b);
+DROP INDEX t10_b ON t10;
+
+CREATE INDEX t10_b ON t10 (b) ALGORITHM=COPY;
+DROP INDEX t10_b ON t10 ALGORITHM=COPY;
+
+# make sure server survives restart
+let $restart_hide_args=1
+let $restart_parameters = restart: --early-plugin-load="keyring_file=$KEYRING_PLUGIN" --loose-keyring_file_data=$MYSQL_TMP_DIR/ts_encrypt_keyring --loose-innodb-temp-tablespace-encrypt $KEYRING_PLUGIN_OPT;
+--source include/restart_mysqld.inc
+
+CREATE TEMPORARY TABLE t01 (a INT) ENGINE=InnoDB;
+INSERT INTO t01 VALUES (1), (2), (3);
+
+# and crash
+let $restart_hide_args=1
+let $restart_parameters = restart: --early-plugin-load="keyring_file=$KEYRING_PLUGIN" --loose-keyring_file_data=$MYSQL_TMP_DIR/ts_encrypt_keyring --loose-innodb-temp-tablespace-encrypt $KEYRING_PLUGIN_OPT;
+--source include/kill_and_restart_mysqld.inc
+
+CREATE TEMPORARY TABLE t01 (a INT) ENGINE=InnoDB;
+INSERT INTO t01 VALUES (1), (2), (3);
+
+# and restart without keyring plugin
+let $restart_hide_args=1
+let $restart_parameters = restart: ;
```

Comment:
consistent spacing around = 
resetting restart_parameters to empty has the same effect as setting it to "restart:"

Reply:
fixed